src/transformers/models/dbrx/configuration_dbrx.py

```diff
@@ -0,0 +1,277 @@
+        moe_jitter_eps (float, optional): The jitter epsilon for the mixture of experts layer.
+        router_aux_loss_coef (float, optional): The loss weight for the mixture of experts layer.
+        moe_normalize_expert_weights (float, optional): The normalization factor for the expert weights.
+        uniform_expert_assignment (bool, optional): Whether to use uniform expert assignment.
+            This should only be used for benchmarking purposes.
+    """
+
+    def __init__(
+        self,
+        ffn_act_fn: Optional[dict] = None,
+        ffn_hidden_size: int = 3584,
+        moe_num_experts: int = 4,
+        moe_top_k: int = 1,
+        moe_jitter_eps: Optional[float] = None,
+        router_aux_loss_coef: float = 0.01,
+        moe_normalize_expert_weights: Optional[float] = 1,
+        uniform_expert_assignment: bool = False,
+        **kwargs: Any,
+    ):
+        super().__init__()
+        if ffn_act_fn is None:
+            ffn_act_fn = {"name": "silu"}
+        self.ffn_act_fn = ffn_act_fn
+        self.ffn_hidden_size = ffn_hidden_size
+        self.moe_num_experts = moe_num_experts
+        self.moe_top_k = moe_top_k
+        self.moe_jitter_eps = moe_jitter_eps
+        self.router_aux_loss_coef = (
+            router_aux_loss_coef if "moe_loss_weight" not in kwargs else kwargs["moe_loss_weight"]
+        )
+        self.moe_normalize_expert_weights = moe_normalize_expert_weights
+        self.uniform_expert_assignment = uniform_expert_assignment
+
+        for k in ["model_type"]:
+            if k in kwargs:
+                kwargs.pop(k)
+        if len(kwargs) != 0:
+            raise ValueError(f"Found unknown {kwargs=}")
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path: str, **kwargs: Any) -> "PretrainedConfig":
+        cls._set_token_in_kwargs(kwargs)
+
+        config_dict, kwargs = cls.get_config_dict(pretrained_model_name_or_path, **kwargs)
+
+        if config_dict.get("model_type") == "dbrx":
+            config_dict = config_dict["ffn_config"]
+
+        if "model_type" in config_dict and hasattr(cls, "model_type") and config_dict["model_type"] != cls.model_type:
+            logger.warning(
+                f"You are using a model of type {config_dict['model_type']} to instantiate a model of type "
+                + f"{cls.model_type}. This is not supported for all configurations of models and can yield errors."
+            )
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class DbrxConfig(PretrainedConfig):
+    r"""
+
+    This is the configuration class to store the configuration of a [`DbrxModel`]. It is used to instantiate a Dbrx model according to the
+    specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the [databricks/dbrx-instruct](https://huggingface.co/databricks/dbrx-instruct) architecture.
```

Comment:
this isn't true right? the defults are for a much smaller model than dbrx.

Reply:
@abhi-mosaic wanted the defaults to be small model

Reply:
Lllama and mixtral both make their default configurations (in `configuration_{model_name}.py`) match their model's actual configurations (in `https://huggingface.co/{model_name}/blob/main/config.json`). See Llama's config [defaults](https://github.com/huggingface/transformers/blob/9b0a8ea7d1d6226b76cfdc645ce65e21157e2b50/src/transformers/models/llama/configuration_llama.py#L118) and its actual [config](https://huggingface.co/meta-llama/Llama-2-7b-chat-hf/blob/main/config.json); see Mixtral's config [defaults](https://github.com/huggingface/transformers/blob/9b0a8ea7d1d6226b76cfdc645ce65e21157e2b50/src/transformers/models/mixtral/configuration_mixtral.py#L114) and its actual [config](https://huggingface.co/mistralai/Mixtral-8x7B-Instruct-v0.1/blob/main/config.json). This is true even for larger models like Mixtral.

It seems like the standard is to make the config defaults of a particular model match the config of that actual model, i.e. we should use the configs of `dbrx-instruct` as our config defaults. However, I also understand that it might be easier to initialize with a smaller model. Happy to put the defaults back to what they were and make them smaller again.

Thoughts @abhi-mosaic?

Reply:
I think we should make the default values a small model because otherwise the nested configs `DbrxAttentionConfig` and `DbrxFFNConfig`, when saved, will recognize that the default values match and **the values will disappear** from the `config.json` that is saved from `.save_pretrained()`. I think this behavior is really bad for readability (things like `kv_n_heads` and `moe_num_experts` would be invisible and you'd have to dig into the `transformers` source to find them). 

To avoid this situation, I think it would be best to make all the default values small and different from the real DBRX. That way `config.json` will be saved out with all values explicitly.

Reply:
Makes sense.

In [this](https://github.com/huggingface/transformers/pull/29921/commits/fb5ed6769f3747ce16d456f189d7d15394e55002) PR, I just put restored the previous default values.

If you want to compare, [this](https://github.com/abhi-mosaic/transformers/commit/efd10b8b64e87352d7e2cbbb8be8139b161702fd) is the PR where I originally made the default configs larger and the same as our model on HF.

Reply:
On this note:

I [just](https://github.com/huggingface/transformers/pull/29921/commits/24b28b5e46db3324d44a22fcba0d485554c92b52) added the parameter `pad_token_id` to our config with the default value of `None`. This is different than `dbrx-instruct` which has `pad_token_id= 100277` ([source](https://huggingface.co/databricks/dbrx-instruct/blob/17365204e9cf13e2296ee984c1ab48071e861efa/added_tokens.json#L4)).

I want to make sure this is the right call here, to make the default `pad_token_id` different than what our model actually uses, correct?

Reply:
hm, that does not seem correct to me. Why should the model config have a padding token? Isn't it just the tokenizer that needs that info?

Reply:
Anyway, you can ignore the above comment.

I thought I needed to add `pad_token_id` to `DbrxConfig` to pass the tests [here](https://github.com/huggingface/transformers/pull/29921#issuecomment-2032584711), but I just found a way around it.

We no longer pass `pad_token_id` into `DbrxConfig`. Instead we just pass `pad_token_id` into `DbrxModelTester` [here](https://github.com/huggingface/transformers/blob/f57e672aa677107e7c7cd07c628b0babbc938c08/tests/models/dbrx/test_modeling_dbrx.py#L36).